src/kakarot/instructions/memory_operations.cairo

```diff
@@ -236,17 +240,47 @@ namespace MemoryOperations {
             );
         }
 
+        // Has to be done BEFORE fetching the current value from the state,
+        // otherwise it would warm up the storage slot.
+        let is_storage_warm = State.is_storage_warm(evm.message.address.evm, key);
+        let account = State.get_account(evm.message.address.evm);
+
+        let original_value = Account.fetch_original_storage(account, key);
+        let current_value = State.read_storage(evm.message.address.evm, key);
+        local gas_cost: felt;
+
+        if (is_storage_warm == FALSE) {
```

Comment:
can we move this right after the `let is_storage_warm` to better split the two chunks of logic